### PR TITLE
Fix ss3migrate

### PR DESCRIPTION
**Pull request by JoshuaCarter**
SS3 Asset Migration Task:
- Make destination directories when required.
- Allow absolute path for src assets dir.
- Improved logging (list any dirs that failed to write to at the end, log individual success/failure, etc).
- Improve comments.
- Improve migration task to work without CDN modules.

### CODE REVIEW

File: code/tasks/SS3MigrateAssetsFromS3.php [hunk 1]
```diff
@@ -16,13 +16,15 @@ public function run($request)
             throw new RuntimeException("This can only be executed from the commandline");
         }
 
-        // The context for the following {src} parameter will be relative to "framework", so "src=../{src}" for example
-        $sourceFolder = trim($request->getVar('src'), "/");
+        // path can be relative or absolute
```
Comment (nglasl) [inline code/tasks/SS3MigrateAssetsFromS3.php]: Would be good to keep a note here regarding the CWD being framework if someone decides to use a relative path.

**Comment by nglasl** [comment]
@JoshuaCarter seems like there's some further changes that can be pulled back here? Just wanted to highlight it so we didn't forget :)

File: code/tasks/SS3MigrateAssetsFromS3.php (hunk 1 above)
Comment (JoshuaCarter) [reply code/tasks/SS3MigrateAssetsFromS3.php]: I've improved the comments a bit more, this is addressed on the class comment block.

**Comment by nglasl** [comment]
No contest from me @JoshuaCarter, I think that's a solid approach. Happy for this to be merged.

**Comment by JoshuaCarter** [comment]
@nglasl done :)

**Pull request merged**